distribution/helm/vector-shared/templates/_metrics.tpl

```diff
@@ -5,40 +5,40 @@ Common Vector configuration partial containing built-in metrics pipeline.
 Internal metrics are common, so we share and reuse the definition.
 */}}
 {{- define "libvector.metricsConfigPartial" -}}
+
 {{- $values := .Values -}}
+
 {{- $prometheusInputs := .prometheusInputs -}}
 {{- with $values.internalMetricsSource }}
 {{- if .enabled }}
 # Emit internal Vector metrics.
-[sources.{{ .sourceId }}]
-  type = "internal_metrics"
-
-  {{- with .rawConfig }}
-  {{- . | nindent 6 }}
-  {{- end }}
+{{- $value := merge (dict) .config -}}
+{{- $_ := set $value "type" "internal_metrics" -}}
+{{- $_ := set $value "rawConfig" .rawConfig -}}
+{{- tuple .sourceId $value | include "libvector.vectorSourceConfig" | nindent 0 -}}
```

Comment:
Couldn't the `| nindent 0` be removed by just removing the initial `-`? I think the same goes for the other uses of `nindent 0`.

```suggestion
{{ tuple .sourceId $value | include "libvector.vectorSourceConfig" -}}
```

Reply:
I didn't check if it could be removed, but to me, it just makes more sense: this form can be read as "remove all the whitespaces before (the `-`s), and prepend with a newline (`nindent 0`)". I've played around with Helm charts for some time now, and I guess this pattern is very useful for defeating the major issues with extra whitespaces.
The `nindent 0` is applied separately from the `-` whitespace removal (that `-` only affects the template), thus allowing for a reliable behaviour no matter how the surrounding template looks - which does not hold when we just rely on the templates without `-`.

The reason I use `nindent 0` here is merely because it *prepends* the string with a newline, a certain printf could also be used here. Helm templates are quite odd for what it's worth, you'd be surprised that a call that prepends something with newline exists usually - but it is what it is.